testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksChangeDataFeedCompatibility.java

```diff
@@ -379,6 +379,71 @@ public void testMergeDeleteIntoTableWithCdfEnabled()
         }
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testMergeMixedDeleteAndUpdateIntoTableWithCdfEnabled()
+    {
+        String targetTableName = "test_merge_mixed_delete_and_update_into_table_with_cdf_" + randomNameSuffix();
+        String sourceTableName = "test_merge_mixed_delete_and_update_into_table_with_cdf_data_table_" + randomNameSuffix();
+        try {
+            onDelta().executeQuery("CREATE TABLE default." + targetTableName + " (page_id INT, page_url STRING, views INT) " +
+                    "USING DELTA " +
+                    "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + targetTableName + "'" +
+                    "TBLPROPERTIES (delta.enableChangeDataFeed = true)");
+            onDelta().executeQuery("CREATE TABLE default." + sourceTableName + " (page_id INT, page_url STRING, views INT) " +
+                    "USING DELTA " +
+                    "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + sourceTableName + "'");
+
+            onDelta().executeQuery("INSERT INTO default." + targetTableName + " VALUES (1, 'pageUrl1', 100)");
+            onDelta().executeQuery("INSERT INTO default." + targetTableName + " VALUES (2, 'pageUrl2', 200)");
+            onDelta().executeQuery("INSERT INTO default." + targetTableName + " VALUES (3, 'pageUrl3', 300)");
+            onDelta().executeQuery("INSERT INTO default." + targetTableName + " VALUES (4, 'pageUrl4', 400)");
+
+            onDelta().executeQuery("INSERT INTO default." + sourceTableName + " VALUES (1000, 'pageUrl1000', 1000)");
+            onDelta().executeQuery("INSERT INTO default." + sourceTableName + " VALUES (2, 'pageUrl2', 20000)");
+            onDelta().executeQuery("INSERT INTO default." + sourceTableName + " VALUES (3000, 'pageUrl3000', 3000)");
+            onDelta().executeQuery("INSERT INTO default." + sourceTableName + " VALUES (4, 'pageUrl4000', 4000)");
+
+            onTrino().executeQuery(
+                    format("""
+                    MERGE INTO delta.default.%s cdf USING delta.default.%s n
+                    ON (cdf.page_id = n.page_id)
+                    WHEN MATCHED AND cdf.page_id = 2
+                        THEN DELETE
+                    WHEN MATCHED AND cdf.page_id > 2
+                        THEN UPDATE SET views = (cdf.views + n.views)
+                    WHEN NOT MATCHED
+                        THEN INSERT (page_id, page_url, views) VALUES (n.page_id, n.page_url, n.views)""",
+                            targetTableName,
+                            sourceTableName));
```

Comment:
nit: Only two variables are used here. How about using string concatenation for SQL syntax highlight in IntelliJ?

Reply:
This is what I had but Marius insisted on Text Block. I also prefer concat